crates/transaction-pool/src/maintain/mod.rs

```diff
@@ -0,0 +1,363 @@
+        client.finalized_block_number().ok().flatten(),
+        canon_processor_config,
+        metrics.clone(),
+    );
+
+    // eviction interval for stale non local txs
+    let mut stale_eviction_interval = time::interval(config.max_tx_lifetime);
+
+    // The update loop that waits for new blocks and reorgs and performs pool updates
+    // Listen for new chain events and derive the update action for the pool
+    loop {
+        trace!(target: "txpool", state=?drift_monitor.state(), "awaiting new block or reorg");
+
+        drift_monitor.update_metrics();
+        let pool_info = pool.block_info();
+
+        // after performing a pool update after a new block we have some time to properly update
+        // dirty accounts and correct if the pool drifted from current state, for example after
+        // restart or a pipeline run
+        if drift_monitor.state().is_drifted() {
+            // assuming all senders are dirty
+            drift_monitor.set_dirty_addresses(pool.unique_senders());
+            // make sure we toggle the state back to in sync
+            drift_monitor.set_state(PoolDriftState::InSync);
+        }
+
+        // if we have accounts that are out of sync with the pool, we reload them in chunks
+        if drift_monitor.has_dirty_addresses() && !drift_monitor.is_reloading() {
+            drift_monitor.start_reload_accounts(
+                client.clone(),
+                pool_info.last_seen_block_hash,
+                &task_spawner,
+            );
+        }
+
+        // check if we have a new finalized block
+        if let Some(BlobStoreUpdates::Finalized(blobs)) = canon_processor.update_finalized(&client)
+        {
+            metrics.inc_deleted_tracked_blobs(blobs.len());
+            // remove all finalized blobs from the blob store
+            pool.delete_blobs(blobs);
+            // and also do periodic cleanup
+            let pool = pool.clone();
+            task_spawner.spawn_blocking(Box::pin(async move {
+                debug!(target: "txpool", "cleaning up blob store");
+                pool.cleanup_blobs();
+            }));
+        }
+
+        // select of account reloads and new canonical state updates which should arrive at the rate
+        // of the block time
+        tokio::select! {
+            // handle reloaded accounts
+            _ = async { futures_util::future::pending::<()>().await }, if !drift_monitor.is_reloading() => {
+                // this branch will never execute, it's just a placeholder
+                // for when we don't have account reloads running
+            }
+            _ = async {}, if drift_monitor.is_reloading() => {
+                if let drift_monitor::DriftMonitorResult::AccountsLoaded(LoadedAccounts { accounts, .. }) = drift_monitor.process_reload_result() {
+                     // update the pool with the loaded accounts
+                     pool.update_accounts(accounts);
+                 }
+            }
```

Comment:
hmm, this looks a bit odd, 

can we keep this like:

https://github.com/paradigmxyz/reth/blob/4e84e42f1e9a285c2e9594ab479cb36b3cd698d1/crates/transaction-pool/src/maintain.rs#L247-L249